libgraph/src/PropertyGraph.cpp

```diff
@@ -921,7 +921,11 @@ katana::PropertyGraph::DeleteNodeIndex(const std::string& column_name) {
       return katana::ResultSuccess();
     }
   }
-  return KATANA_ERROR(katana::ErrorCode::NotFound, "node index not found");
+
+  // TODO(Chak-Pong) make deleteNodeIndex always successful
+  //  before index existence check is available from python side
+  //  return KATANA_ERROR(katana::ErrorCode::NotFound, "node index not found");
+  return katana::ResultSuccess();
```

Comment:
I don't think it is a good idea to always return true as it loses semantics - why return anything at all? If the signature returns a `Result` it should have some information.
Secondly, i think deleting an index is an important enough operation to require acknowledgement that the index was indeed deleted. The alternate is to add a `HasIndex` call after deletion to verify if the index was deleted. This way a single call can be used to check if the index did exist and was indeed deleted.

Reply:
Yes, I agree except that we might not be able to perform the existence check along with the edit(create/delete index) within one single call because the current return type is `void`.

I think We either need to change the return type to be `bool` or expose the existence check to python.  If the former is preferred, I can do it in the follow-up PR.

Reply:
If the goal is to switch this back soon, then this should definitely have  warning `KATANA_LOG_WARNING` when it does this.